### PR TITLE
add top bar with volume, mute state, radio name and current track

### PR DESCRIPTION
**Pull request by slashformotion**
image

I had to change quite a lot of code to get a top bar especially because the list model was made to take the whole window by default

### CODE REVIEW

**Comment by slashformotion** [comment]
I also changed the maximum volume to 150 because it is actually the max of libmpv

File: gui/list.go [hunk 1]
```diff
@@ -55,6 +57,7 @@ func UpdateMetadataLoop(m *model, delayToRefresh time.Duration) {
 func (m *model) Init() tea.Cmd {
 	m.playingItemIndex = m.list.Index()
 	m.volume = 100
+	width = tm.Width()
```
Comment (DucNg) [inline gui/list.go]: Don't use external lib to get terminal width, you should be able to use [WindowSizeMsg.Width](https://pkg.go.dev/github.com/charmbracelet/bubbletea#WindowSizeMsg.Width) from bubbletea

File: gui/list.go [hunk 2]
```diff
@@ -98,17 +97,17 @@ func (m *model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 			m.metadataLoopChan <- struct{}{} // Stop the existing loop
 			go UpdateMetadataLoop(m, 0)      // Start the new one
 		case "+", "-":
-			if msg.String() == "+" && m.volume < 100 {
+			if msg.String() == "+" && m.volume < 150 {
```
Comment (DucNg) [inline gui/list.go]: Setting max volume to 100 was a deliberate choice to avoid distortions

File: gui/list.go [hunk 3]
```diff
@@ -167,7 +175,7 @@ func Render(ins *dbus.Instance, mpv *player.MPV, lastRadioIndex int) int {
 func setMetadata(m *model) time.Duration {
 	playingItem := m.list.Items()[m.playingItemIndex].(item)
 	fm := metadata.FetchMetadata(playingItem.metadataUrl)
-
+	m.trackName = fm.Now.FirstLine
```
Comment (DucNg) [inline gui/list.go]: We could also add artistName with SecondLine?

**Review by DucNg — COMMENTED** [review]
Very pretty good job! If possible I would like to avoid using external libs apart from bubbletea

File: gui/list.go (hunk 2 above)
Comment (slashformotion) [reply gui/list.go]: done

File: gui/list.go (hunk 3 above)
Comment (slashformotion) [reply gui/list.go]: infos are now formated `<track> - <artist>`

**Comment by slashformotion** [comment]
image

**Comment by DucNg** [comment]
Stange things happen when name is too long, probably not linked to anything you've done so fine for now, we'll need to fix that later tho:
image

**Review by DucNg — APPROVED** [review]
Very nice and pretty thanks! I'll push some minor changes but good to me!

**Pull request merged**